geopandas/tests/test_geoseries.py

```diff
@@ -331,6 +331,10 @@ def test_from_xy_points(self):
         assert_geoseries_equal(
             self.landmarks, GeoSeries.from_xy(x, y, index=index, crs=crs)
         )
+        assert_geoseries_equal(
+            self.landmarks,
+            GeoSeries.from_xy(self.landmarks.x, self.landmarks.y, crs=crs),
+        )
```

Comment:
Can you also add a test case with `z` argument. And maybe also a test case where the index is _not_ equal?

Reply:
A good suggestion, thanks! I added the tests.